### PR TITLE
Update the bootstrapper images for k8s 1.21

### LINKED ISSUE #11424: Repository and tag for CoreDNS changed in 1.8.0 (Kubernetes 1.21)
`Error response from daemon: manifest for k8s.gcr.io/coredns:1.8.0 not found: manifest unknown: Failed to fetch "1.8.0" from request "/v2/coredns/manifests/1.8.0".`

https://github.com/kubernetes/kubernetes/commit/40862c3b6045ba198562a0f238dc2bb20758a361

```diff
--- a/cmd/kubeadm/app/constants/constants.gopkg/minikube/bootstrapper/images/images.go
+++ b/cmd/kubeadm/app/constants/constants.go
@@ -323,7 +323,7 @@ const (
        CoreDNSDeploymentName = "coredns"
 
        // CoreDNSImageName specifies the name of the image for CoreDNS add-on
-       CoreDNSImageName = "coredns"
+       CoreDNSImageName = "coredns/coredns"
 
        // KubeDNSConfigMap specifies in what ConfigMap in the kube-system namespace the kube-dns config should be stored
        KubeDNSConfigMap = "kube-dns"
@@ -344,7 +344,7 @@ const (
        KubeDNSVersion = "1.14.13"
 
        // CoreDNSVersion is the version of CoreDNS to be deployed if it is used
-       CoreDNSVersion = "1.7.0"
+       CoreDNSVersion = "v1.8.0"
 
        // ClusterConfigurationKind is the string kind value for the ClusterConfiguration struct
        ClusterConfigurationKind = "ClusterConfiguration"

```

So this change needs to be reflected in the minikube code as well.

```go
func coreDNS(v semver.Version, mirror string) string
```

### LINKED ISSUE #11425: Pause version changed in Kubernetes 1.21
It was bumped from "3.2" (on Unix) to "3.4.1"

https://github.com/kubernetes/kubernetes/commit/8946f2676983266a819de1292444a538acf5c2fe

```diff
diff --git a/cmd/kubeadm/app/constants/constants.go b/cmd/kubeadm/app/constants/constants.go
index 57ddbcc5764..228ef196553 100644
--- a/cmd/kubeadm/app/constants/constants.go
+++ b/cmd/kubeadm/app/constants/constants.go
@@ -414,6 +414,9 @@ const (
        ModeRBAC string = "RBAC"
        // ModeNode is an authorization mode that authorizes API requests made by kubelets.
        ModeNode string = "Node"
+
+       // PauseVersion indicates the default pause image version for kubeadm
+       PauseVersion = "3.4.1"
 )
 
 var (
diff --git a/cmd/kubeadm/app/constants/constants_unix.go b/cmd/kubeadm/app/constants/constants_unix.go
index b024348c29a..16ff72d5556 100644
--- a/cmd/kubeadm/app/constants/constants_unix.go
+++ b/cmd/kubeadm/app/constants/constants_unix.go
@@ -21,7 +21,4 @@ package constants
 const (
        // DefaultDockerCRISocket defines the default Docker CRI socket
        DefaultDockerCRISocket = "/var/run/dockershim.sock"
-
-       // PauseVersion indicates the default pause image version for kubeadm
-       PauseVersion = "3.2"
 )
diff --git a/cmd/kubeadm/app/constants/constants_windows.go b/cmd/kubeadm/app/constants/constants_windows.go
index e8b2494490a..6daae0a1fff 100644
--- a/cmd/kubeadm/app/constants/constants_windows.go
+++ b/cmd/kubeadm/app/constants/constants_windows.go
@@ -21,7 +21,4 @@ package constants
 const (
        // DefaultDockerCRISocket defines the default Docker CRI socket
        DefaultDockerCRISocket = "npipe:////./pipe/docker_engine"
-
-       // PauseVersion indicates the default pause image version for kubeadm
-       PauseVersion = "1.4.0"
 )

```

So this change needs to be reflected in the minikube code as well.

```go
func Pause(v semver.Version, mirror string) string
```

----

Note: this function is public, because it is copied to the configuration for the CRI.

Currently it is still hardcoded in dims [cri-dockerd](https://github.com/dims/cri-dockerd), as `defaultPodSandboxImageVersion`

But updated in kubernetes/pkg/kubelet/dockershim (or github.com/Mirantis/cri-dockerd):

```go
        defaultSandboxImage = "k8s.gcr.io/pause:3.4.1"
```

### PR DESCRIPTION
**Pull request by afbjorklund**
The list of images had not been updated for 1.21, since the release was pinned at 1.20

Add a unit test case to make sure that the output matches what `kubeadm` is expecting.

Remove support for anything older than 1.12

The oldest tested version is currently 1.14

Closes #11424
Closes #11425

### CODE REVIEW

File: pkg/minikube/bootstrapper/images/images.go [hunk 1]
```diff
@@ -29,8 +29,12 @@ import (
 // Pause returns the image name to pull for a given Kubernetes version
 func Pause(v semver.Version, mirror string) string {
 	// Should match `PauseVersion` in:
+	// https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/constants/constants.go
+	pv := "3.4.1"
```
Comment (medyagh) [inline pkg/minikube/bootstrapper/images/images.go:33]: this means we will need to bump the preload version
https://github.com/medyagh/minikube/blob/1be59b8f23c66d9a6be174dcd579b17302857f5f/pkg/minikube/download/preload.go#L45

lets also comment somewhere in this file that bump preload version if u change this versions

**Review by medyagh — CHANGES_REQUESTED** [review]

File: pkg/minikube/bootstrapper/images/images.go (hunk 1 above)
Comment (afbjorklund) [reply pkg/minikube/bootstrapper/images/images.go:33]: I think it only changed for Kubernetes 1.21.0, but maybe you had already generated preloads for that with the wrong images ?

Comment (afbjorklund) [reply pkg/minikube/bootstrapper/images/images.go:33]: Indeed, preloaded-images-k8s-v10-v1.21.0-docker-overlay2-amd64.tar.lz4 has the wrong pause and the wrong coredns

Comment (afbjorklund) [reply pkg/minikube/bootstrapper/images/images.go:33]: the previous versions are all correct though, and I'm not sure we built anything for 1.22 yet ? (I added "master", not alpha)

**Comment by medyagh** [comment]
the unit test failure is not because of this PR
we need to fix the unit test not to fail when there is no preload  https://github.com/kubernetes/minikube/issues/11439
```
--- FAIL: TestDownload (0.82s)
    --- FAIL: TestDownload/PreloadDownloadPreventsMultipleDownload (0.22s)
        download_test.go:94: Failed to download preload: open /home/runner/.minikube/cache/preloaded-tarball/preloaded-images-k8s-v11-v1.20.2-docker-overlay2-amd64.tar.lz4.690737788: no such file or directory
            tempfile
            k8s.io/minikube/pkg/minikube/download.Preload
            	/home/runner/work/minikube/minikube/pkg/minikube/download/preload.go:163
            k8s.io/minikube/pkg/minikube/download.testPreloadDownloadPreventsMultipleDownload.func4
            	/home/runner/work/minikube/minikube/pkg/minikube/download/download_test.go:93
            runtime.goexit
            	/opt/hostedtoolcache/go/1.16.4/x64/src/runtime/asm_amd64.s:1371
        download_test.go:94: Failed to download preload: open /home/runner/.minikube/cache/preloaded-tarball/preloaded-images-k8s-v11-v1.20.2-docker-overlay2-amd64.tar.lz4.470237611: no such file or directory
            tempfile
            k8s.io/minikube/pkg/minikube/download.Preload
            	/home/runner/work/minikube/minikube/pkg/minikube/download/preload.go:163
            k8s.io/minikube/pkg/minikube/download.testPreloadDownloadPreventsMultipleDownload.func4
            	/home/runner/work/minikube/minikube/pkg/minikube/download/download_test.go:93
            runtime.goexit
            	/opt/hostedtoolcache/go/1.16.4/x64/src/runtime/asm_amd64.s:1371
        download_test.go:105: Expected only 1 download attempt but got 0!
FAIL
```

**Review by medyagh — APPROVED** [review]

**Pull request merged**